### DISCUSSION TITLE
I have seen Adguard MV3 Beta allowing custom filters but why does UBlock Origin Lite does not?

### DISCUSSION THREAD

**Discussion by thedoggybrad** [uBlockOrigin/uBOL-home#153] [Q&A]
Just curious,
I have seen Adguard MV3 Beta allowing custom filters but why does UBlock Origin Lite does not?

Any chance of custom filters becoming a reality in UBlock Origin Lite?

Note: I do not have any technical knowledge regarding the full concept of both MV2 and MV3.

**Comment by stephenhawk8054** [comment]
You can read here: https://github.com/uBlockOrigin/uBOL-home/wiki/Frequently-asked-questions-(FAQ)#is-ubo-lite-a-bad-faith-attempt-at-converting-ubo-to-mv3

> My goal with uBO Lite is a **reliable** and **efficient** MV3-compliant content blocker, leveraging the filter lists used by uBO. I am satisfied that I fulfilled that goal by ensuring uBO Lite was entirely declarative -- though at the cost of limitations beyond those intrinsic to MV3.

> **Reliability:** Being entirely declarative, uBO Lite's service worker is not required to initiate DNR-, cosmetic-, or scriptlet-based filtering in a timely manner.
> 
> As a result, uBO Lite reliably filters at browser launch, or when navigating to new webpages while its service worker is suspended. This can't be achieved without uBO Lite's declarative approach.

> The choice of being entirely declarative for the sake of reliability and efficiency meant sacrificing being able to create custom filters, hence the "Lite" in uBO Lite: it's not meant as an MV3-compliant version of uBO, it's meant as a reliable Lite version of uBO, suitable for those who used uBO in an install-and-forget manner.

---

Example:

<details><summary>Adguard MV3 when launching Chromium and restoring the last youtube page</summary>

https://github.com/user-attachments/assets/1978419d-fa91-4a9c-8c5f-2a86cb92ff17

</details>

<details><summary>uBO Lite when launching Chromium and restoring the last youtube page</summary>

https://github.com/user-attachments/assets/5259e7dc-f487-4b2d-a2d5-23e5675024a8

</details>

**Reply by thedoggybrad** [reply]
Thanks to your reply. Helped me a lot and decided to migrate from UBlock Origin (MV2) to UBlock Origin Lite. You proved that UBlock Origin Lite is still usable without custom filters.

**Reply by garry-ut99** [reply]
Don't spam, like mentioned here: [LINK 1](https://github.com/uBlockOrigin/uBOL-home/issues/127#issuecomment-2186606571) + [LINK 2](https://github.com/uBlockOrigin/uBOL-home/issues/127#issuecomment-2188492690), search for duplicates before posting, as your issue is another duplicate of a pile of already existing duplicates, same or similiar questions are to be asked in already existing threads:
- https://github.com/uBlockOrigin/uBOL-home/issues/17
- https://github.com/uBlockOrigin/uBOL-home/discussions/37
- https://github.com/uBlockOrigin/uBOL-home/issues/36
- https://github.com/uBlockOrigin/uBOL-home/discussions/88
- https://github.com/uBlockOrigin/uBOL-home/issues/125
- https://github.com/uBlockOrigin/uBOL-home/issues/127
- https://github.com/uBlockOrigin/uBOL-home/issues/134

> Thanks to your reply. Helped me a lot 

Steven's reply is mainly what already exists in documentation/ Wikipedia, and my reply is mainly what already exists in issue tracker. Read documentation/Wikipedia and search issue tracker for duplicates, before spamming. uBlock volunteers are not on your call, particularly to answer the same / similiar questions over and over again, optionally, if you still wanted to ask your question, you should do so in already existing threads.

**Comment by nascentt** [comment]
Pity you cant as there's no filterlist I'm aware of that can stop the artificial delays on youtube which can be blocked on UBO with `||googlevideo.com/videoplayback$xhr,3p,method=get,domain=www.youtube.com`
having to put up with the constant delays on youtube because that filter cant be added is a real pity.

**Reply by stephenhawk8054** [reply]
Enable `Developer mode` in uBOL settings -> Go to `Develop` pane -> Select `Custom DNR rules` and copy the above filter there.

**Reply by Yuki2718** [reply]
Maybe we can add the rule to the Experimental filter.

**Reply by stephenhawk8054** [reply]
That filter would cause issue for some users. There were some reports back then.

I think we should use regex for it, based on their fake datetime values.

**Reply by stephenhawk8054** [reply]
@Yuki2718 I think this would be quite safe for main list too:

```adb
/\.googlevideo\.com\/videoplayback\?expire=(?:[02-9]+|1[1-68-9]\d+|17[1-48-9]\d+)&/$xhr,3p,method=get,domain=www.youtube.com
```

The concern would not really be about the breakages, but whether they are applying detection based on those requests or not.

**Reply by Yuki2718** [reply]
Are they? I don't remember them to detect these blocking.

**Reply by stephenhawk8054** [reply]
Yeah, I don't remember they use detection with it either. Just a bit careful.

But other than that, I think the filter is quite safe.

**Reply by stephenhawk8054** [reply]
@gorhill Wait, `method` doesn't work in uBOL?

---

Ah nope, it works but when I copy the above filter to uBOL, `get` becomes upper-case and shows error. I had to adjust it to lower-case

<img width="2520" height="1608" alt="image" src="https://github.com/user-attachments/assets/74e00133-ec2c-4b05-8a74-d819ea9473c1" />

**Reply by gorhill** [reply]
It should work with `get`. I will update the uBOL DNR converter to normalize casing to lower.

---

https://github.com/gorhill/uBlock/commit/b9956a8d187d2aa3dda1ad94ce7fa805752061c7

**Reply by stephenhawk8054** [reply]
I added the filter for fake `googlevideo`: https://github.com/uBlockOrigin/uAssets/commit/84b8f44eade54acf32bdb0d1413298861eae4d74
